arroyo/processing/strategies/dead_letter_queue/policies/abstract.py

```diff
@@ -1,18 +1,50 @@
 from abc import ABC, abstractmethod
-from typing import Any, Sequence
+from datetime import datetime
+from typing import Optional, Sequence, Union
 
+Serializable = Union[str, bytes]
 
-class InvalidMessages(Exception):
+
+class InvalidMessage(Exception):
     """
-    An exception to be thrown to pass bad messages to the DLQ
-    so they are handled correctly.
+    An exception to be thrown to pass a bad message to the DLQ.
+
+    A reason can be optionally passed describing why the message
+    is bad.
+
+    If the original topic the message was produced to is known,
+    that should be passed along via this exception.
     """
 
-    def __init__(self, messages: Sequence[Any]):
-        self.messages = messages
+    def __init__(
+        self,
+        message: Serializable,
+        reason: Optional[str] = None,
+        original_topic: Optional[str] = None,
+    ):
+        self.message = message
+        self.reason = reason or "unknown"
+        self.topic = original_topic or "unknown"
+        self.timestamp = str(datetime.now())
 
     def __str__(self) -> str:
-        return f"Invalid Message(s): {self.messages}"
+        return (
+            f"Invalid Message originally produced to: {self.topic}\n"
+            f"Reason: {self.reason}\n"
+            f"Exception thrown at: {self.timestamp}\n"
+            f"Message: {str(self.message)}"
+        )
+
+
+class InvalidBatchedMessages(Exception):
+    """
+    An exception to be thrown to pass batched invalid messages.
+    to the DLQ. Generally should only be used by batching
+    strategies which collect multiple `InvalidMessage` exceptions.
+    """
+
+    def __init__(self, exceptions: Sequence[InvalidMessage]):
+        self.exceptions = exceptions
```

Comment:
I don't see the reason for this exception. I think your approach in PR https://github.com/getsentry/arroyo/pull/57 is better. Have one exception and that can contain multiple payloads, one per bad message. 
I don't see a reason for representing an individual bad message as an exception itself.

Reply:
I'll refactor this (or maybe merge that in here?) to follow the same approach in that PR once that is finalized.